components/PropsTable.tsx

```diff
@@ -0,0 +1,124 @@
+import docgen from '@atom-learning/components/dist/docgen.json'
+import * as React from 'react'
+
+import { styled } from '../stitches.config'
+import { Box, InlineCode, Link, Text } from './'
+
+type PropsTableProps = {
+  for: React.FC
+}
+
+const columns = ['Prop', 'Type', 'Default value', 'Required']
```

Comment:
Just wondering, how we would visualise in the table the different values for the restricted properties. I have seen some components that have a column for that.

However, we could just document all the different options in the documentation itself and thus not need them here

Reply:
What do you mean by restricted properties?

Reply:
for instance, for `Text` we have a size prop that can only be `sm`, `md` or `lg` or in `Button` the appearance prop can only be `solid` and `outline`

Reply:
Ah ok, that's an interesting thought. We can definitely look to tweak this structure based on our prop patterns for these components, and we have the data to be able to retain a separate field for `type` and then list the available options separately. I think this is worth reviewing when we have completed more components and have a better idea of the props that we'll need to list